Kernel/Syscalls/fallocate.cpp

```diff
@@ -12,11 +12,13 @@
 
 namespace Kernel {
 
+// https://pubs.opengroup.org/onlinepubs/9699919799/functions/posix_fallocate.html
 ErrorOr<FlatPtr> Process::sys$posix_fallocate(int fd, Userspace<off_t const*> userspace_offset, Userspace<off_t const*> userspace_length)
 {
     VERIFY_NO_PROCESS_BIG_LOCK(this);
     TRY(require_promise(Pledge::stdio));
 
+    // [EINVAL] The len argument is less than zero, or the offset argument is less than zero, or the underlying file system does not support this operation.
```

Comment:
Can we write this comment as `// NOTE: Return EINVAL if the...`?

Reply:
These comments are copied exactly from the spec. I think it's more valuable to have exactly the same text as the spec, instead of a slightly reworded versions. This is what we already do with other code in the project when specs are referenced.

Reply:
Seems OK to me. Maybe we could invent a format for this
`// RETURN VALUE: [EINVAL] ...`
which let you easily copy the text from the spec but still find it easily with an easy grep?